qa/tasks/persistent_write_log_cache_thrash.py

```diff
@@ -0,0 +1,99 @@
+"""
+persistent write log cache thrash task
+"""
+import contextlib
+import logging
+import random
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.orchestra import run
+
+DEFAULT_KILL_DAEMON_TIME = 20
+io_pattern = ("seq", "rand")
+io_size = ('4', '16', '128', '1024')
+
+log = logging.getLogger(__name__)
+
+@contextlib.contextmanager
+def thrashes_rbd_bench_on_persistent_cache(ctx, config):
+    """
+    thrashes rbd bench on persistent write log cache.
+    It can test recovery feature of persistent write log cache.
+    """
+    log.info("thrashes rbd bench on persistent write log cache")
+
+    client, client_config = list(config.items())[0]
+    (remote,) = ctx.cluster.only(client).remotes.keys()
+    client_config = client_config if client_config is not None else dict()
+    image_name = client_config.get('image_name', 'testimage')
+    cache_path = client_config.get('rbd_persistent_cache_path', '/home/ubuntu/cephtest/rbd-pwl-cache')
+    kill_daemon_time = client_config.get('kill_daemon_time', DEFAULT_KILL_DAEMON_TIME)
+    ls_command = '"$(ls {} )"'.format(cache_path)
```

Comment:
I would remove `ls` variant and do both checks based on `rbd status`.  This would also conveniently get rid of `rbd_persistent_cache_path` parameter that you are currently duplicating from `home/4-cache-path.yaml` as the cache path wouldn't be needed.

Reply:
The `rbd status` command still reads the state stored in the ceph cluster. Since the process of `rbd bench` is killed, there is naturally no application to change the state in the cluster. It will be consistent with the previous `rbd status`. But `ls` tests whether there are dirty cache files locally after `rbd bench` being killed. Please consider this. If you insist, I will make a change.

Reply:
So, can I keep both of these checks? Just like the current code.

Reply:
If you want to keep the `ls` check, I would suggest:
- parse the cache path out of `rbd status` output (add `--format=json` use `json.loads()` to make that easier)
- get rid of `rbd_persistent_cache_path` parameter